pkg/rke2/rke2_linux.go

```diff
@@ -56,16 +58,187 @@ func initExecutor(clx *cli.Context, cfg Config, dataDir string, disableETCD bool
+				}
+			}
+		}
+	}
+
+	if cfg.ControlPlaneResourceLimits != "" {
+		for _, r := range strings.Split(cfg.ControlPlaneResourceLimits, ",") {
+			v := strings.Split(r, "=")
+			if len(v) != 2 {
+				logrus.Fatalf("incorrectly formatted control plane resource limit specified: %s", r)
+			}
+			if strings.HasPrefix(v[0], KubeAPIServer) {
+				resource := strings.TrimPrefix(v[0], KubeAPIServer+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.KubeAPIServerCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.KubeAPIServerMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+			if strings.HasPrefix(v[0], KubeScheduler) {
+				resource := strings.TrimPrefix(v[0], KubeScheduler+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.KubeSchedulerCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.KubeSchedulerMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+			if strings.HasPrefix(v[0], KubeControllerManager) {
+				resource := strings.TrimPrefix(v[0], KubeControllerManager+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.KubeControllerManagerCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.KubeControllerManagerMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+			if strings.HasPrefix(v[0], KubeProxy) {
+				resource := strings.TrimPrefix(v[0], KubeProxy+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.KubeProxyCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.KubeProxyMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+			if strings.HasPrefix(v[0], Etcd) {
+				resource := strings.TrimPrefix(v[0], Etcd+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.EtcdCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.EtcdMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+			if strings.HasPrefix(v[0], CloudControllerManager) {
+				resource := strings.TrimPrefix(v[0], CloudControllerManager+"-")
+				switch resource {
+				case "cpu":
+					controlPlaneResources.CloudControllerManagerCPULimit = v[1]
+				case "memory":
+					controlPlaneResources.CloudControllerManagerMemoryLimit = v[1]
+				default:
+					logrus.Fatalf("unrecognized resource limit made: %s", r)
+				}
+			}
+		}
+	}
+
+	extraEnv := podexecutor.ControlPlaneEnv{
+		KubeAPIServer:          cfg.ExtraEnv.KubeAPIServer.Value(),
+		KubeScheduler:          cfg.ExtraEnv.KubeScheduler.Value(),
+		KubeControllerManager:  cfg.ExtraEnv.KubeControllerManager.Value(),
+		Etcd:                   cfg.ExtraEnv.Etcd.Value(),
+		CloudControllerManager: cfg.ExtraEnv.CloudControllerManager.Value(),
+	}
+
+	extraMounts := podexecutor.ControlPlaneMounts{
```

Comment:
Should we do input validation on these? Later on we do `mount := strings.Split(rawMount, ":")` and reference `mount[1]` but we don't actually validate that they contain at least one semicolon, so that could cause a runtime panic.

Reply:
Yeah, I've added some basic input validation to check that the length of the resultant slice is 2, but this doesn't actually validate that the paths are valid or anything. I'm not sure how far we want to go with the validations, as the kubelet will block the static pods from running itself as well if the volume mounts/source don't make sense